website/docs/r/app_service_environment_v3.html.markdown

```diff
@@ -0,0 +1,116 @@
+---
+subcategory: "App Service (Web Apps)"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_app_service_environment"
+description: |-
+  Manages an App Service Environment V3.
+
+---
+
+# azurerm_app_service_environment
+
+Manages an App Service Environment V3.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "exampleRG1"
+  location = "West Europe"
+}
+
+resource "azurerm_virtual_network" "example" {
+  name                = "example-vnet1"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  address_space       = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet" "ase" {
+  name                 = "asesubnet"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefixes     = ["10.0.1.0/24"]
+}
+
+resource "azurerm_subnet" "gateway" {
+  name                 = "gatewaysubnet"
+  resource_group_name  = azurerm_resource_group.example.name
+  virtual_network_name = azurerm_virtual_network.example.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_app_service_environment_v3" "example" {
+  name      = "example-ase"
+  subnet_id = azurerm_subnet.ase.id
+
+  cluster_setting {
+    name  = "DisableTls1.0"
+    value = "1"
+  }
+
+  cluster_setting {
+    name  = "InternalEncryption"
+    value = "true"
+  }
+
+  cluster_setting {
+    name  = "FrontEndSSLCipherSuiteOrder"
+    value = "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384_P256,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256_P256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384_P256,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256_P256,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA_P256,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA_P256"
+  }
+
+  tags = {
+    env         = "production"
+    terraformed = "true"
+  }
+
+}
+
+```
+
+## Argument Reference
+
+* `name` - (Required) The name of the App Service Environment. Changing this forces a new resource to be created. 
+
+* `resource_group_name` - (Required) The name of the Resource Group where the App Service Environment exists. Defaults to the Resource Group of the Subnet (specified by `subnet_id`).
+
+* `subnet_id` - (Required) The ID of the Subnet which the App Service Environment should be connected to. Changing this forces a new resource to be created.
+
+~> **NOTE** a /24 or larger CIDR is required. Once associated with an ASE, this size cannot be changed.
+
+* `cluster_setting` - (Optional) Zero or more `cluster_setting` blocks as defined below. 
+
+* `tags` - (Optional) A mapping of tags to assign to the resource. 
```

Comment:
Tags are not part of the update model, and don't appear to be present in the swagger... This _could_ be an oversight, but this behaviour is also present in V1 and V2.

Reply:
> This PR introduces the preview resource, App Service Environment V3.
> 
> SDK/API coverage is incomplete, so two notable items are missing, but will be added if/when they are supported. (These are `Host group deployment`, and `upgrade preference`.)
> 
> supersedes #10304
> Resolves #9420

All is there a thought this would make it into v2.57.0?   Seems all buttoned up. Would love to get our hands on this to get ahead of a pending GA date.